### PR TITLE
avoid pip packages

### PR DESCRIPTION
**Pull request by ocefpaf**
Similar to https://github.com/pangeo-data/pangeo-cloud-federation/pull/305

### CODE REVIEW

File: pangeo-notebook/binder/environment.yml [hunk 1]
```diff
@@ -56,6 +58,3 @@ dependencies:
   # xarray-related
   - xgcm
   - xrft
-  - pip:
-    - regionmask
-    - rio-cogeo
```
Comment (jhamman) [inline pangeo-notebook/binder/environment.yml]: Thanks for doing this @ocefpaf. One quick request? This image inherits from the base-notebook image. Can you remove the pip installs there too?

Comment (ocefpaf) [reply pangeo-notebook/binder/environment.yml]: > One quick request? This image inherits from the base-notebook image. Can you remove the pip installs there too?

Sure. Can you point me where that is? I'm still familiarizing myself with the pangeo repos. (Also, I'm planning to add the `strict` channel option and add the defaults channel back but that must be done on the Dockerfile. See my explanation in https://github.com/pangeo-data/pangeo-cloud-federation/pull/305.)

Comment (jhamman) [reply pangeo-notebook/binder/environment.yml]: see https://github.com/pangeo-data/pangeo-stacks/blob/master/base-notebook/binder/environment.yml#L16

Comment (ocefpaf) [reply pangeo-notebook/binder/environment.yml]: Done.

PS: where is the canonical place for the Dockerfile? I would like to add the `strict` option and maybe `pip` in the conda root env so the `%pip` magic works in the notebooks.

Comment (jhamman) [reply pangeo-notebook/binder/environment.yml]: Thanks @ocefpaf - the docker file addition will need to some special consideration so let's deal with that separately.

Comment (ocefpaf) [reply pangeo-notebook/binder/environment.yml]: Yep. I'm planning 2-3 PRs to implement everything in a controlled way.

**Pull request merged**